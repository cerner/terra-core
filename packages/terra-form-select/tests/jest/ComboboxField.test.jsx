import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import ComboboxField from '../../src/ComboboxField';

it('should render ComboboxField component with max height', () => {
  const select = (
    <ComboboxField label="Label" defaultValue="blue" selectId="select-id" maxHeight={100}>
      <ComboboxField.Option value="blue" display="Blue" />
      <ComboboxField.Option value="green" display="Green" />
      <ComboboxField.Option value="purple" display="Purple" />
      <ComboboxField.Option value="red" display="Red" />
      <ComboboxField.Option value="violet" display="Violet" />
    </ComboboxField>
  );

  expect(shallowWithIntl(select).dive()).toMatchSnapshot();
});

