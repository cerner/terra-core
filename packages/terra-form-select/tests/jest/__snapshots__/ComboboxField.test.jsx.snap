// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComboboxField component with max height 1`] = `
<div
  className="field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div
    className="label-group"
  >
    <label
      className="label"
      htmlFor="select-id"
    >
      Label
    </label>
    <div
      className="error-icon-hidden"
    >
      <IconError
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
  <InjectIntl(Combobox)
    allowClear={false}
    ariaLabel="Label"
    defaultValue="blue"
    disabled={false}
    inputId="select-id"
    isIncomplete={false}
    isInvalid={false}
    isTouchAccessible={false}
    key=".0"
    maxHeight={100}
    required={false}
  >
    <Option
      disabled={false}
      display="Blue"
      value="blue"
    />
    <Option
      disabled={false}
      display="Green"
      value="green"
    />
    <Option
      disabled={false}
      display="Purple"
      value="purple"
    />
    <Option
      disabled={false}
      display="Red"
      value="red"
    />
    <Option
      disabled={false}
      display="Violet"
      value="violet"
    />
  </InjectIntl(Combobox)>
</div>
`;
